aten/src/ATen/native/cpu/RangeFactoriesKernel.cpp

```diff
@@ -0,0 +1,37 @@
+#include <cmath>
+#include <ATen/Config.h>
+#include <ATen/Dispatch.h>
+
+#include <ATen/cpu/vec256/vec256.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+
+
+namespace at { namespace native {
+namespace {
+
+using namespace vec256;
+
+static void linspace_kernel(TensorIterator& iter, Scalar scalar_start, Scalar scalar_end, int64_t steps) {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(iter.dtype(), "linspace_cpu", [&]() {
+    int64_t idx(0);
+    scalar_t start = scalar_start.to<scalar_t>();
+    scalar_t end = scalar_end.to<scalar_t>();
+    scalar_t step = (end - start) / static_cast<scalar_t>(steps - 1);
+    cpu_kernel_vec(
```

Comment:
You need to use serial kernel here if you want to modify `idx`.

Reply:
Thanks. I tried using `atomic` yesterday but it seems less efficient than the current implementation. I'll try serial_kernel later on.

Reply:
I switched to `cpu_serial_kernel` and updated the benchmark. I was expecting no perf difference, but surprised that the new code is magically more efficient for float.